src/uu/base32/src/base_common.rs

```diff
@@ -143,25 +143,48 @@ pub fn base_app(about: &'static str, usage: &str) -> Command {
         )
 }
 
-pub fn get_input(config: &Config) -> UResult<Box<dyn Read>> {
+/// A trait alias for types that implement both `Read` and `Seek`.
+pub trait ReadSeek: Read + Seek {}
+
+/// Automatically implement the `ReadSeek` trait for any type that implements both `Read` and `Seek`.
+impl<T: Read + Seek> ReadSeek for T {}
+
+pub fn get_input(config: &Config) -> UResult<Box<dyn ReadSeek>> {
     match &config.to_read {
         Some(path_buf) => {
             // Do not buffer input, because buffering is handled by `fast_decode` and `fast_encode`
             let file =
                 File::open(path_buf).map_err_context(|| path_buf.maybe_quote().to_string())?;
-
             Ok(Box::new(file))
         }
         None => {
-            let stdin_lock = io::stdin().lock();
-
-            Ok(Box::new(stdin_lock))
+            let mut buffer = Vec::new();
+            io::stdin().read_to_end(&mut buffer)?;
+            Ok(Box::new(io::Cursor::new(buffer)))
         }
     }
 }
 
-pub fn handle_input<R: Read>(input: &mut R, format: Format, config: Config) -> UResult<()> {
-    let supports_fast_decode_and_encode = get_supports_fast_decode_and_encode(format);
+/// Determines if the input buffer ends with padding ('=') after trimming trailing whitespace.
+fn has_padding<R: Read + Seek>(input: &mut R) -> UResult<bool> {
```

Comment:
It looks a bit odd to have a bool function that returns a `UResult`. Though currently I don't see how this could be improved :|

Reply:
yeah, agreed :)
i tried to return a bool but we would not get the error mgmt on "input.read_to_end"